src/vmm/src/lib.rs

```diff
@@ -354,13 +354,29 @@ impl Vmm {
 
     /// Sets RDA bit in serial console
     pub fn emulate_serial_init(&self) -> Result<()> {
+        #[cfg(target_arch = "aarch64")]
+        use devices::legacy::SerialDevice;
         #[cfg(target_arch = "x86_64")]
         let mut serial = self
             .pio_device_manager
             .stdio_serial
             .lock()
             .expect("Poisoned lock");
 
+        #[cfg(target_arch = "aarch64")]
```

Comment:
we can have a cfg-ed block, like so:
```
#[cfg(target_arch = "aarch64")]
{
... code here
}
```

Reply:
turns out it's very tricky to get this right due to lifetime constraints. let's keep things as they are